ml-agents/mlagents/trainers/optimizer/torch_optimizer.py

```diff
@@ -49,25 +55,115 @@ def create_reward_signals(self, reward_signal_configs):
                 reward_signal, self.policy.behavior_spec, settings
             )
 
+    def _evaluate_by_sequence(
+        self, tensor_obs: List[torch.Tensor], initial_memory: np.ndarray
+    ) -> Tuple[Dict[str, torch.Tensor], AgentBufferField, torch.Tensor]:
+        """
+        Evaluate the batch sequence-by-sequence, assembling the result. This enables us to get the
+        intermediate memories for the critic.
```

Comment:
describe the inputs and their dimensions. it is not clear what initial_memory's shape is

Reply:
Added a comment